src/sentry/tasks/relay.py

```diff
@@ -34,9 +35,20 @@ def build_project_config(public_key=None, **kwargs):
     try:
         from sentry.models import ProjectKey
 
+        now = time.time()
         sentry_sdk.set_tag("public_key", public_key)
         sentry_sdk.set_context("kwargs", kwargs)
 
+        metrics.incr(
+            "relay.projectconfig_cache.run",
+            tags={"task": "build"},
+            sample_rate=1,
+        )
+        schedule_duration = now - kwargs.get("tmp_scheduled", now)
```

Comment:
I don't think falling back to `now` is a good idea, I'd rather not emit any metrics if we don't get `tmp_scheduled` in the params, or fallback to `0` (or `-1` if it's possible) to know we are missing some cases.

Reply:
I believe the fallback is only there to bridge the gap during the deploy, where old tasks are processed by new workers.

Reply:
yes, it is only during deploys and having them show up as 0 is reasonably nice: they're still counted but cleerly not true (well, unless the clock drift between machines means we sometimes hit real 0.  I also considered defaulting to something huge like say 500s but didn't purely because it'd mean these few falling through here would distort the graph too much (datadog doesn't allow you to set the Y-axis limits i think)